### PR TITLE
initial work on DatabaseVersion#isAfter

### PR DESCRIPTION
**Pull request by sebersole**
Opening as draft as I wanted to get feedback about the contracts here prior to changing `Dialect#getVersion` and its uses

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/dialect/DatabaseVersion.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+	 * Determine whether this version before the passed one
+	 */
+	default boolean isBefore(Integer major, Integer minor) {
+		return isBefore( (int) major, minor == null ? NO_VERSION : minor );
+	}
+
+	/**
+	 * Determine whether this version before the passed one
+	 */
+	default boolean isBefore(int otherMajor, int otherMinor, int otherMicro) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/dialect/DatabaseVersion.java:238]: I would prefer the name `isAtLeast` but YMMV

**Review by gavinking — APPROVED** [review]
Yup, all LGTM.

**Review by beikov — APPROVED** [review]

**Review by dreab8 — APPROVED** [review]
LGTM :+1:

File: hibernate-core/src/main/java/org/hibernate/dialect/DatabaseVersion.java (hunk 1 above)
Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/dialect/DatabaseVersion.java:238]: As an alternative to isBefore?  I assume you mean isSince?

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/dialect/DatabaseVersion.java:238]: Yeah. Apparently I clicked on the wrong line

**Comment by sebersole** [comment]
Applied upstream

File: hibernate-core/src/main/java/org/hibernate/dialect/DatabaseVersion.java (hunk 1 above)
Comment (sebersole) [reply hibernate-core/src/main/java/org/hibernate/dialect/DatabaseVersion.java:238]: I landed on `#isSameOrAfter`

**Pull request closed without merging**